File: backend/app/models/comment.rb
class Comment < ApplicationRecord
  belongs_to :task
  belongs_to :user

  validates :content, presence: true, length: { maximum: 2_000 }

  def api_json
    {
      id: id,
      task_id: task_id,
      content: content,
      user: user_json,
      created_at: created_at&.iso8601,
      updated_at: updated_at&.iso8601
    }
  end

  private

  def user_json
    {
      id: user.id,
      name: user.name
    }
  end
end

File: frontend/src/types/comment.ts
import type { UserMini } from './task'

export type Comment = {
  id: number
  task_id: number
  content: string
  user: UserMini
  created_at: string
  updated_at: string
}

export type CommentsResponse = {
  comments: Comment[]
}

export type CommentResponse = {
  comment: Comment
}

export type CreateCommentInput = {
  content: string
}
